compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -59,8 +59,17 @@ object Applications {
     extractorMemberType(tp, nme.get, errorPos).exists
 
   def productSelectorTypes(tp: Type, errorPos: Position = NoPosition)(implicit ctx: Context): List[Type] = {
-    val sels = for (n <- Iterator.from(0)) yield extractorMemberType(tp, nme.selectorName(n), errorPos)
-    sels.takeWhile(_.exists).toList
+    def tupleSelectors(n: Int, tp: Type): List[Type] = {
+      val sel = extractorMemberType(tp, nme.selectorName(n), errorPos)
+      if (sel.exists) sel :: tupleSelectors(n + 1, tp) else Nil
```

Comment:
Maybe add a comment to explain why sel can be NoType here

Reply:
added